src/python/pants/fs/archive.py

```diff
@@ -41,7 +51,7 @@ class TarArchiver(Archiver):
   """
 
   @classmethod
-  def extract(cls, path, outdir):
+  def _extract(cls, path, outdir):
     """
     :API: public
```

Comment:
Because this method is `:API: public`, an equivalent method needs to be exposed (although you can definitely add parameters that have defaults). It looks like probably the `:API: public` tag should move to the super class, and the `def _extract` methods should have it removed. Not sure how to handle the `filter_func` argument... maybe just pass kwargs through the super class method to the `_extract` impl?

Reply:
I updated the documentation.  We could implement filter_func for tars as well if it is necessary to unify the API.